src/dash/DashHandler.js

```diff
@@ -352,8 +352,8 @@ function DashHandler(config) {
 
         index = getIndexForSegments(time, representation, timeThreshold);
         //Index may be -1 if getSegments needs to update.  So after getSegments is called and updated then try to get index again.
-        updateSegments(representation);
         if (index < 0) {
+            updateSegments(representation);
```

Comment:
I do not think this will work for all cases unfortunately.  I would have to test to be sure but index is only -1 at start and updateSegment is key to be updated many times for dynamic streams especially if not segment template.   What type of MPD are you working with on this ?